plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.cdi.reddeer.validators;
+
+import java.util.Arrays;
+
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView.ProblemType;
+import org.jboss.tools.cdi.reddeer.annotation.ValidationType;
+
+public class BeanValidationProviderCDI20 extends AbstractValidationProvider{
+	
+	private static final String JSR = "JSR-365"; 
+	private static final String DEL_ANNOTATION_INJECT_FROM_METHOD = "Delete annotation @Inject from method";
+	private static final String DEL_ANNOTATION_OBSERVES_FROM_PARAMETER = "Delete annotation @Observes from parameter";
+	private static final String DEL_ANNOTATION_DISPOSES_FROM_PARAMETER = "Delete annotation @Disposes from parameter";
+	private static final String DEL_ANNOTATION_PRODUCES_FROM_METHOD = "Delete annotation @Produces from method";
+	
+	@Override
+	void init() {
+		problems.add(new ValidationProblem(ProblemType.WARNING, ValidationType.NO_BEAN_ELIGIBLE,
```

Comment:
Same messages as in CDI11 validation provider.

Reply:
Yes, the messages are the same.

Reply:
We agreed, right? So, what shall we do with a repeatable code?

Reply:
Hint: we have three classes with same content where only one variable differs and is derived based on CDI version used (1.0, 1.1. 2.0). How can we define the content only once and still has JSR value based on version used?

Reply:
Fixed.